abra_core/src/parser/parser.rs

```diff
@@ -164,7 +164,10 @@ impl Parser {
             Token::Eq(_) | Token::Neq(_) => Precedence::Equality,
             Token::GT(_) | Token::GTE(_) | Token::LT(_) | Token::LTE(_) => Precedence::Comparison,
             Token::Assign(_) => Precedence::Assignment,
-            Token::Dot(_) | Token::LBrack(_) | Token::LParen(_) => Precedence::Call,
+            Token::Dot(_) | Token::LParen(_) => Precedence::Call,
+            Token::LBrack(_, is_preceded_by_newline) => {
+                if *is_preceded_by_newline { Precedence::None } else { Precedence::Call }
```

Comment:
This is where the "magic" happens - where we're checking to see whether the LBrack (which is currently being treated as a possible infix token) has a higher or lower precedence than what's currently being parsed. If it's preceded by a newline, then we know to treat that as an array literal expression, not an index operator.